src/middleware/serverMiddlewareLoader.middleware.ts

```diff
@@ -4,35 +4,57 @@ import bodyParser from 'body-parser'
 import { createLoggerMiddleware } from 'ch-logging'
 import ApplicationLogger from 'ch-logging/lib/ApplicationLogger'
 import cookieParser from 'cookie-parser'
-import * as express from 'express'
+import { Application, NextFunction, Request, RequestHandler, Response } from 'express'
 import { INTERNAL_SERVER_ERROR } from 'http-status-codes'
 import { inject } from 'inversify'
 import { provide } from 'inversify-binding-decorators'
 import NunjucksLoader from './nunjucksLoader.middleware'
 
 import { APP_NAME } from 'app/constants/app.const'
+import initContainer from 'app/inversify.config'
+import SessionService from 'app/services/session/session.service'
+import TYPES from 'app/types'
 
 @provide(ServerMiddlewareLoader)
 export default class ServerMiddlewareLoader {
 
   public constructor(
     @inject(NunjucksLoader) private nunjucks: NunjucksLoader,
-    @inject(ApplicationLogger) private logger: ApplicationLogger) {}
+    @inject(ApplicationLogger) private logger: ApplicationLogger,
+    @inject(SessionService) private sessionService: SessionService
+  ) {}
 
-  public loadServerMiddleware(app: express.Application, directory: string): void {
+  public loadServerMiddleware(app: Application, directory: string): void {
     app.use(bodyParser.json())
     app.use(bodyParser.urlencoded({ extended: true }))
     app.use(cookieParser())
     app.use(createLoggerMiddleware(APP_NAME))
 
+    const sessionMiddleware: RequestHandler = initContainer().get(TYPES.SessionMiddleware)
+    app.use(sessionMiddleware)
+
+    this.saveAuthenticatedUserEmailToLocalVariable(app)
+
     this.nunjucks.configureNunjucks(app, directory)
   }
 
-  public configureErrorHandling(app: express.Application): void {
-    app.use((err: any, _: express.Request, res: express.Response, _2: express.NextFunction) => {
+  public configureErrorHandling(app: Application): void {
+    app.use((err: any, _: Request, res: Response, _2: NextFunction) => {
       this.logger.error(`${err.constructor.name} - ${err.message}`)
 
       return res.status(err.statusCode || INTERNAL_SERVER_ERROR).render('error')
     })
   }
+
+  private saveAuthenticatedUserEmailToLocalVariable(app: Application): void {
+    app.use((req: Request, res: Response, next: NextFunction) => {
+      if (req.session) {
+        const userEmail: string = this.sessionService.getUserEmail(req)
+
+        res.locals.userEmail = userEmail
+      }
+
+      return next()
+    })
```

Comment:
I would move this out to it's own middleware file if possible. Let's keep this file as light as possible. Also let's just do:
```
res.locals.userEmail = this.sessionService.getUserEmail(req)
```

to save a couple of lines